src/ILCompiler.Compiler/src/Compiler/SingleFileCompilationModuleGroup.cs

```diff
@@ -27,7 +27,11 @@ public override void AddCompilationRoots(IRootingServiceProvider rootProvider)
         {
             base.AddCompilationRoots(rootProvider);
 
-            AddCompilationRootsForExports((EcmaModule)_typeSystemContext.SystemModule, rootProvider);
+            foreach (var inputFile in _typeSystemContext.ReferenceFilePaths)
+            {
```

Comment:
Hm. Interesting. The main reason I brought up this change was because in my private-repo project I had a number of assemblies that needed to be scanned for `NativeCallableAttribute`. I only used `ReferenceFilePaths` because that was what worked for me at the time I first wrote it. If you want to go with `InputFilePaths` instead, it would easy enough for me to adjust my build scripts.